From 1c52de742e8ce35ebea51fca54842d50757220d7 Mon Sep 17 00:00:00 2001
From: Xing Zheng <zhengxing@rock-chips.com>
Date: Tue, 17 Jul 2018 21:02:31 +0800
Subject: [PATCH 48/73] ASoC: rk3308_codec: fix incorrect idx with loopback_grp

Here should use the mapped grp as a reference, not idx.

Reported-by: Lin Jianhua <linjh@rock-chips.com>
Signed-off-by: Xing Zheng <zhengxing@rock-chips.com>
(cherry picked from commit 7ea44331974b7d00553aab24376044105d70ed55)

Change-Id: Id0b10701e62a6ec31ab26fb9fa9be127cde4dae5
Signed-off-by: Xing Zheng <zhengxing@rock-chips.com>
---
 sound/soc/codecs/rk3308_codec.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/sound/soc/codecs/rk3308_codec.c b/sound/soc/codecs/rk3308_codec.c
index 889dce93ef87..7d73a1fee55c 100644
--- a/sound/soc/codecs/rk3308_codec.c
+++ b/sound/soc/codecs/rk3308_codec.c
@@ -588,7 +588,7 @@ static bool adc_for_each_grp(struct rk3308_codec_priv *rk3308,
 
 		if ((!rk3308->loopback_dacs_enabled) &&
 		    has_loopback(rk3308->loopback_grp) &&
-		    rk3308->loopback_grp == idx) {
+		    rk3308->loopback_grp == mapped_grp) {
 			/*
 			 * Ths loopback DACs are closed, and specify the
 			 * loopback ADCs.
-- 
2.11.0

